### DISCUSSION TITLE
Autocommit should always be disabled by default

### DISCUSSION THREAD

**Discussion by heyciao** [dbeaver/dbeaver#13046] [Q&A]
Hi

Current situation:
autocommit is enabled for all connections tagged as non-"Production".

Problem nbr1:
the "connection type" is often ignored by users during the setup of the connection (e.g. no warning stating that "autocommit" will be on), thinking that it's just a "label" or something similar (I personally didn't even notice that categorization).

Problem nbr2:
different behaviour for people that transition between DEV and Prod Support roles.

Proposed fix:
have autocommit always disabled.

Details:
I initially tried to push the company that I'm working for to use DBeaver, but having "autocommit" switched on by default is suicide (even for me).
I'm permanently super-stressed, same for my colleagues, so paying attention to the association to the connection type (Development/Test/Production) for the hundreds of DBs that we have, when the connection is set up, is unrealistic.
Even for development/test I would personally not like any default-autocommit-behaviour.

Personal opinion:
- autocommit-by-default was maybe ok in your early days to keep support effort low by avoiding stupid questions (e.g. "my data is not being stored", etc...).
- if you want to be accepted as a serious tool, having autocommit switched on by default (for any environment) is like shooting yourself in the foot.

Thank you

**Comment by p91paul** [comment]
you can also change default autocommit setting for all connection types in global preferences

**Comment by ShadelessFox** [comment]
Also, you can switch connection type to `Production`.

https://github.com/dbeaver/dbeaver/wiki/Connection-Types

**Comment by mtwig** [comment]
I share the opinion of the discussion author.

While it's possible to change the global default for dev connections, I think the danger comes from the default connection type being dev, and the default setting for dev being autocommit.

It's very easy to not notice that the connection types are more than a label, so likely many people will ignore it. There's documentation in a wiki, but I doubt many people will read the wiki before opening the tool and trying it out. 

That leaves a very real risk that someone will create a new connection, which gets created in the default connection type (Development), and the default auto-commit setting (on). 

It would likely be safer to ensure the _default_ settings are "safe", and if someone wants to have auto-commit on by default for dev connections, they're able to opt into that dangerous default. 

An alternative "solution" to this problem could be to at least assume new connections are production by default - so at least some sort of action is required to make a connection "Development". But I also can't relate to the desire to ever using autocommit at all, let-alone by default.
